kubectl describe endpointslice panic when an Endpoint doesn't have zone

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


#### What happened:
`kubectl describe endpointslice` panic as below when an Endpoint doesn't have zone.
```
# kubectl describe endpointslice -n kube-system
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x17b8d43]

goroutine 1 [running]:
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe.describeEndpointSliceV1.func1(0x207ff00, 0xc000240370, 0xc000eb82a0, 0x0)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe/describe.go:3059 +0x7a3
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe.tabbedString(0xc00074fa70, 0x0, 0x0, 0x0, 0x0)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe/describe.go:5132 +0xbc
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe.describeEndpointSliceV1(0xc001150c60, 0xc000b868c0, 0x2085e30, 0xc001150c60, 0xc000b868c0, 0x0)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe/describe.go:2991 +0x5c
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe.(*EndpointSliceDescriber).Describe(0xc0002f7430, 0xc000f96850, 0xb, 0xc000f96824, 0xc, 0x1, 0x0, 0x0, 0x1c00e00, 0xc0006a1aa0)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/describe/describe.go:2975 +0x105
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/describe.(*DescribeOptions).Run(0xc000840f00, 0x0, 0x1f064f0)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/describe/describe.go:188 +0x4ee
k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/describe.NewCmdDescribe.func1(0xc00083e840, 0xc000511a40, 0x1, 0x3)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/describe/describe.go:110 +0x9f
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc00083e840, 0xc0005119e0, 0x3, 0x3, 0xc00083e840, 0xc0005119e0)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:854 +0x2c2
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0004f2dc0, 0xc000204120, 0xc00020c000, 0x5)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:958 +0x375
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:895
main.main()
        _output/dockerized/go/src/k8s.io/kubernetes/cmd/kubectl/kubectl.go:49 +0x21d
```

#### What you expected to happen:
`kubectl describe endpointslice` should print zone as `<unset>` if an Endpoint's zone is unset.

#### How to reproduce it (as minimally and precisely as possible):
1. Do not set "topology.kubernetes.io/zone" for Nodes.
2. Create Pods and export them as a service.
3. Use kubectl to describe the corresponding EndpointSlice.

#### Anything else we need to know?:

#### Environment:
- Kubernetes version (use `kubectl version`): v1.21.0
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

--- a/staging/src/k8s.io/kubectl/pkg/describe/describe.go
+++ b/staging/src/k8s.io/kubectl/pkg/describe/describe.go
@@ -3055,7 +3055,7 @@ func describeEndpointSliceV1(eps *discoveryv1.EndpointSlice, events *corev1.Even
 				w.Write(LEVEL_2, "NodeName:\t%s\n", nodeNameText)
 
 				zoneText := "<unset>"
-				if endpoint.NodeName != nil {
+				if endpoint.Zone != nil {
 					zoneText = *endpoint.Zone
 				}
 				w.Write(LEVEL_2, "Zone:\t%s\n", zoneText)

--- a/staging/src/k8s.io/kubectl/pkg/describe/describe_test.go
+++ b/staging/src/k8s.io/kubectl/pkg/describe/describe_test.go
@@ -4874,6 +4874,7 @@ Events:         <none>` + "\n",
 						Addresses:  []string{"1.2.3.6", "1.2.3.7"},
 						Conditions: discoveryv1.EndpointConditions{Ready: utilpointer.BoolPtr(true)},
 						TargetRef:  &corev1.ObjectReference{Kind: "Pod", Name: "test-124"},
+						NodeName:   utilpointer.StringPtr("node-2"),
 					},
 				},
 				Ports: []discoveryv1.EndpointPort{
@@ -4906,7 +4907,7 @@ Endpoints:
       Ready:    true
     Hostname:   <unset>
     TargetRef:  Pod/test-124
-    NodeName:   <unset>
+    NodeName:   node-2
     Zone:       <unset>
 Events:         <none>` + "\n",
 		},

